axon-framework/axon-framework-commands/infrastructure.md

```diff
@@ -414,59 +376,242 @@ public DisruptorCommandBus commandBus(TransactionManager txManager, AxonConfigur
 
-Configuring a distributed command bus can \(mostly\) be done without any modifications in configuration files.
+{% tabs %}
+{% tab title="AnnotationRoutingStrategy" %}
+```java
+// A custom annotation can be used to drive the AnnotationRoutingStrategy
+@interface CustomRoutingAnnotation {
+}
+// Constructs a AnnotationRoutingStrategy with a different annotation and fallback: 
+public RoutingStrategy routingStrategy() {
+    return AnnotationRoutingStrategy.builder()
+                                    .annotationType(CustomRoutingAnnotation.class)
+                                    .fallbackRoutingStrategy(UnresolvedRoutingKeyPolicy.STATIC_KEY)
+                                    .build();
+}
+```
+{% endtab %}
+
+{% tab title="MetaDataRoutingStrategy" %}
+```java
+// Constructs a MetaDataRoutingStrategy with the defined meta data key and a different fallback:
+public RoutingStrategy routingStrategy() {
+    return MetaDataRoutingStrategy.builder()
+                                  .metaDataKey("my-routing-key")
+                                  .fallbackRoutingStrategy(UnresolvedRoutingKeyPolicy.RANDOM_KEY)
+                                  .build();
+}
+```
+{% endtab %}
+{% endtabs %}
+
+Of course a custom implementation of the `RoutingStrategy` can also be provided when necessary
+When we need to deviate from the default `AnnotationRoutingStrategy`, we should configure it like so: 
+
+{% tabs %}
+{% tab title="Axon Configuration API" %}
+```java
+public class AxonConfig {
+    // ...  
+    public void configureRoutingStrategy(Configurer configurer, RoutingStrategy routingStrategy) {
+        configurer.registerComponent(RoutingStrategy.class, config -> routingStrategy);
+    }
+}
+```
+{% endtab %}
 
-First of all, the starters for one of the Axon distributed command bus modules needs to be included \(e.g. [JGroups](../../extensions/jgroups.md) or [Spring Cloud](../../extensions/spring-cloud.md)\).
+{% tab title="Spring Boot AutoConfiguration" %}
+```java
+@Configuration
+public class AxonConfig {
+    // ...
+    @Bean
+    public RoutingStrategy routingStrategy() {
+        return /* construct your routing strategy */;
+    }
+}
+```
+{% endtab %}
+{% endtabs %}
+
+### AxonServerCommandBus
+
+The `AxonServerCommandBus` is the _default_ distributed `CommandBus` implementation set by the framework.
+It connects to [AxonServer](../../axon-server-introduction.md) to submit and receive commands with.
 
-Once that is present, a single property needs to be added to the application context, to enable the distributed command bus:
+As it is the default, configuring it is relatively straightforward:
 
+{% tabs %}
+{% tab title="Axon Configuration API" %}
+Declare dependencies:
 ```text
-axon.distributed.enabled=true
+<!--somewhere in the POM file-->
+<dependency>
+    <groupId>org.axonframework</groupId>
+    <artifactId>axon-server-connector</artifactId>
+    <version>${axon.version}</version>
```

Comment:
Should we start using `axon.bom.version` for example? just trying to figure out the best way to approach the bom on the docs as well (as I am getting used to that still)

Reply:
Hmmm true...in essence, the BOM allows removal of the version here entirely, but it requires adding a dependency management section too which I would like to omit actually.

Reply:
I've simply expanded on the sample like so:
```xml
<!-- somewhere in the POM file... -->
<dependencyManagement>
    <!-- amongst the dependencies... -->
    <dependencies>
        <dependency>
            <groupId>org.axonframework</groupId>
            <artifactId>axon-bom</artifactId>
            <version>${version.axon}</version>
            <type>pom</type>
            <scope>import</scope>
        </dependency>
    </dependencies>
</dependencyManagement>
<!-- ... -->
<dependencies>
    <!-- amongst the dependencies... -->
    <dependency>
        <groupId>org.axonframework</groupId>
        <artifactId>axon-server-connector</artifactId>
    </dependency>
    <dependency>
        <groupId>org.axonframework</groupId>
        <artifactId>axon-configuration</artifactId>
    </dependency>
</dependencies>
```

Reply:
Looks good!